src/main/java/io/tiledb/java/api/Range.java

```diff
@@ -0,0 +1,236 @@
+package io.tiledb.java.api;
+
+import io.tiledb.libtiledb.*;
+import java.math.BigInteger;
+
+public class Range implements AutoCloseable {
+  private tiledb_range_t range;
+  private Domain domain;
+  private Context ctx;
+  private Long dimIdx;
+  private NativeArray min, max;
+
+  /**
+   * Constructor
+   *
+   * @param ctx The TileDB context
+   * @param domain The array domain
+   * @param dimIdx The dimension index to add the range to
+   */
+  public Range(Context ctx, Domain domain, long dimIdx) {
+    this.domain = domain;
+    this.ctx = ctx;
+    this.range = new tiledb_range_t();
+    this.dimIdx = dimIdx;
+  }
+
+  /**
+   * Constructor
+   *
+   * @param ctx The TileDB context
+   * @param range C pointer to the range
+   * @param domain The array domain
+   * @param dimIdx The dimension index to add the range to
+   */
+  public Range(Context ctx, tiledb_range_t range, Domain domain, long dimIdx) {
+    this.ctx = ctx;
+    this.range = range;
+    this.domain = domain;
+    this.dimIdx = dimIdx;
+  }
+
+  /**
+   * Sets the min value for this range
+   *
+   * @param min The min value
+   * @throws TileDBError
+   */
+  public synchronized void setMin(Object min) throws TileDBError {
+    Datatype dimType;
+
+    dimType = domain.getDimension(dimIdx).getType();
+
+    // We use java type check here because we can not tell the difference between unsigned and
+    // signed
+    // values coming from java, i.e. A UINT16 and INT32 are both Integer classes in java.
+    Types.javaTypeCheck(min.getClass(), dimType.javaClass());
+
+    NativeArray minArr = new NativeArray(ctx, 1, dimType);
+    minArr.setItem(0, min);
+    if (!dimType.isStringType()) {
+      setMinSize(BigInteger.valueOf(dimType.getNativeSize()));
+    }
+    range.setMin(minArr.toVoidPointer());
+  }
+
+  /**
+   * Sets the min size
+   *
+   * @param size The size
+   */
+  public synchronized void setMinSize(long size) {
```

Comment:
I'm not sure what the reason for adding `synchronized` is, but you can remove it from the wrapper methods.

Reply:
Done!